illumos-utils/src/zfs.rs

```diff
@@ -949,24 +1034,90 @@ mod test {
     #[test]
     fn parse_dataset_missing_fields() {
         let expect_missing = |input: &str, what: &str| {
-            let err = DatasetProperties::from_str(input)
+            let err = DatasetProperties::parse_many(input)
                 .expect_err("Should have failed to parse");
             let err = format!("{err:#}");
             assert!(err.contains(&format!("Missing {what}")), "{err}");
         };
 
         expect_missing(
-            "-       dataset_name        1234   5678   111      222",
-            "'compression'",
+            "dataset_name used 5678 -\n\
+             dataset_name quota 111 -\n\
+             dataset_name reservation 222 -\n\
+             dataset_name compression off inherited",
+            "'available'",
         );
         expect_missing(
-            "-       dataset_name        1234   5678   111",
-            "'reservation'",
+            "dataset_name available 1234 -\n\
+             dataset_name quota 111 -\n\
+             dataset_name reservation 222 -\n\
+             dataset_name compression off inherited",
+            "'used'",
+        );
+        expect_missing(
+            "dataset_name available 1234 -\n\
+             dataset_name used 5678 -\n\
+             dataset_name quota 111 -\n\
+             dataset_name reservation 222 -",
+            "'compression'",
         );
-        expect_missing("-       dataset_name        1234   5678", "'quota'");
-        expect_missing("-       dataset_name        1234", "'used'");
-        expect_missing("-       dataset_name", "'avail'");
-        expect_missing("-", "'name'");
-        expect_missing("", "UUID");
+    }
+
+    #[test]
+    fn parse_dataset_uuid_ignored_if_inherited() {
```

Comment:
This test explicitly covers the case we are observing from inventory right now.

Because this UUID is inherited, it should be ignored.